proparse/src/main/java/org/prorefactor/proparse/antlr4/Lexer.java

```diff
@@ -0,0 +1,1117 @@
+        case 's':
+        case 't':
+        case 'u':
+        case 'v':
+        case 'w':
+        case 'x':
+        case 'y':
+        case 'z':
+          append();
+          getChar();
+          ttype = PreprocessorParser.FILENAME;
+          break;
+        case '.':
+          if (prepro.isNameDot()) {
+            append();
+            getChar();
+            break;
+          } else
+            break for_loop;
+        default:
+          break for_loop;
+      }
+    }
+    if (currText.length() == 1)
+      return makeToken(inputType);
+    else
+      return makeToken(ttype);
+  }
+
+  ProToken periodStart() throws IOException {
+    LOGGER.trace("Entering periodStart()");
+
+    if (!Character.isDigit(currChar)) {
+      if (prepro.isNameDot())
+        return makeToken(PreprocessorParser.NAMEDOT);
+      else
+        return makeToken(PreprocessorParser.PERIOD);
+    }
+    int ttype = PreprocessorParser.NUMBER;
+    for_loop : for (;;) {
+      switch (currChar) {
+        // We don't know here if the plus or minus is in the middle or at the end.
+        // Don't change _ttype.
+        case '+':
+        case '-':
+        case '0':
+        case '1':
+        case '2':
+        case '3':
+        case '4':
+        case '5':
+        case '6':
+        case '7':
+        case '8':
+        case '9':
+          append();
+          getChar();
+          break;
+        case '#':
+        case '$':
+        case '%':
+        case '&':
+        case '/':
+        case '\\':
+        case '_':
+        case 'a':
+        case 'b':
+        case 'c':
+        case 'd':
+        case 'e':
+        case 'f':
+        case 'g':
+        case 'h':
+        case 'i':
+        case 'j':
+        case 'k':
+        case 'l':
+        case 'm':
+        case 'n':
+        case 'o':
+        case 'p':
+        case 'q':
+        case 'r':
+        case 's':
+        case 't':
+        case 'u':
+        case 'v':
+        case 'w':
+        case 'x':
+        case 'y':
+        case 'z':
+          append();
+          getChar();
+          ttype = PreprocessorParser.FILENAME;
+          break;
+        default:
+          break for_loop;
+      }
+    }
+    return makeToken(ttype);
+  }
+
+  ProToken id(int inputTokenType) throws IOException {
+    LOGGER.trace("Entering id()");
+
+    // Tokens that start with a-z or underscore
+    // - ID
+    // - FILENAME
+    // - keyword (testLiterals = true)
+    // Also inputTokenType can be ANNOTATION for a token that starts with '@'.
+    // Based on the PROGRESS online help, the following are the valid name characters.
+    // Undocumented: you can use a slash in an index name! Arg!
+    // Undocumented: the compiler allows you to start a block label with $
+    // If we find a back slash, we know we're into a filename.
+    // Extended characters (octal 200-377) can be used in identifiers, even at the beginning.
+    int ttype = inputTokenType;
+    for_loop : for (;;) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor the code to remove this label and the need for it. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3ALabelsShouldNotBeUsedCheck)